src/django/pfb_analysis/management/commands/add_city_fips.py

```diff
@@ -0,0 +1,145 @@
+from django.core.management.base import BaseCommand
+
+import csv
+import logging
+import os
+import sys
+
+import boto3
+from botocore.exceptions import ClientError
+import fiona
+from fiona.crs import from_epsg
+
+logger = logging.getLogger(__name__)
+
+FIPS_CSV = 'national_places.csv'
+# local path to cached FIPS CSV
+FIPS_PATH = '/data/{csv}'.format(csv=FIPS_CSV)
+# S3 bucket where FIPS CSV should exist
+FIPS_BUCKET = 'pfb-public-documents'
+
+
+def add_city_fips(input_path, output_path, fips_csv):
+    logger.info("""Writing city FIPS Shapefile for {input}
+                to {output} using FIPS from {fips}""".format(input=input_path,
+                                                             output=output_path,
+                                                             fips=fips_csv))
+    # ensure Unicode will be handled properly
+    reload(sys)
+    sys.setdefaultencoding('utf8')
+
+    # map states to place names and their FIPS, for fast lookup
+    lookup = {}
+    with open(FIPS_PATH) as inf:
+        rdr = csv.DictReader(inf)
+        for place in rdr:
+            if not (place['STATE']) in lookup:
+                fips = '{state}{place}'.format(state=place['STATEFP'], place=place['PLACEFP'])
+                lookup[place['STATE']] = {place['PLACENAME'].upper(): fips}
+            else:
+                lookup[place['STATE']][place['PLACENAME'].upper()] = fips
+
+    with fiona.open(input_path) as shp:
+        schema = shp.schema.copy()
+        # CRS not set on input, so explicitly set it here
+        crs = from_epsg(4326)
+        # Add field to Shapefile properties
+        schema['properties']['city_fips'] = 'str:7'
+        with fiona.open(output_path, 'w', driver='ESRI Shapefile',
+                        schema=schema, crs=crs) as outshp:
+            place_count = len(shp)
+            found = 0
+            for feature in shp:
+                prop = feature.get('properties')
+                city = prop['city'].upper()
+                state_places = lookup.get(prop['state']).keys()
+                found_fips = None
+                if state_places:
+                    for place in state_places:
+                        if place.find(city) > -1:
```

Comment:
```suggestion
                        if place.find(city) == 0:
```
The file I was trying had Sioux City in it, which matches both "Sioux City" and "South Sioux City" with this check.  Matching only at the beginning resolves that one, though it wouldn't help with "Grand Rapids" vs. "Grand Rapids charter township".  Not that this has to be perfect, but we might as well get as close as we conveniently can.